### PR TITLE
Microverse > Project Calculator > Milestone 4 > FB Logic

### PR DESCRIPTION
**Pull request by jurgen1c**
## In this branch:

- Add calculate method
- Add operate method
- Import calculate to App

### CODE REVIEW

File: src/logic/calculate.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  let nextNum = next;
+
+  if (bName === '+/-') {
+    result *= -1;
+    nextNum *= -1;
+    return { result, nextNum };
+  }
+  result = operate(result, nextNum, bName);
+  return { result, nextNum };
+};
```
Comment (trekab) [inline src/logic/calculate.js]: The `calculator data object` should contain three properties: `total`, `next` and `operation` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone4/totalNextOperate.png)). You seem to have ignored the 3rd property in your logic.

File: src/logic/operate.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      break;
+    case '*':
+      result = numOne * numTwo;
+      break;
+    case '÷':
+      result = numOne / numTwo;
+      break;
+    case '%':
+      result = numOne % numTwo;
+      break;
```
Comment (trekab) [inline src/logic/operate.js]: `%` should turn the number into a percentage based number, `e.g. 95 % == .95` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone4/performOperate.png))

**Review by trekab — CHANGES_REQUESTED** [review]
### Changes Required! :recycle: 
Well done @jurgen1c! Nice job working on most of the requirements for this milestone. There is a small issue for you to look into before finishing this milestone though. 

- [ ] Check out the comment in the code below about the arguments you are expected to have and use for the `calculate` function.
- [ ] Check out the `%` logic such that it returns the expected output as indicated in the comment below.

__Kindly make the necessary changes and submit another code review request.__
_Feel free to reach me on slack for any inquiries about this review._ :wink: 
__Treasure Kabareebe__
[Portfolio](https://trekab.github.io/portfolio/) | [LinkedIn](https://www.linkedin.com/in/treasure-kabareebe/) | [Twitter](https://twitter.com/TKabareebe)

**Review by BrittanyBlake — APPROVED** [review]
### Status: Approved ✅ 
****
### Hi @jurgen1c! 👋 
### Great job implementing the requested changes!! 🎖️ It's a 👍 from me! Go ahead and merge this pull request and keep up the good work 😄
****


 Good Luck and Happy Coding! 🍀 👩🏾‍💻

**Pull request merged**